airflow/models/taskinstance.py

```diff
@@ -709,6 +709,12 @@ def _execute_task(task_instance: TaskInstance | TaskInstancePydantic, context: C
         execute_callable = task_to_execute.resume_execution
         execute_callable_kwargs["next_method"] = task_instance.next_method
         execute_callable_kwargs["next_kwargs"] = task_instance.next_kwargs
+    elif task_instance.next_method == TaskDeferred.TRIGGER_EXIT:
```

Comment:
I'm not sure we'll ever reach this `elif`. If `elif task_instance.next_method == TaskDeferred.TRIGGER_EXIT`, shouldn't it match `if task_instance.next_method` on line 704 first?